### PR TITLE
Pre-populate RenewRegistration with the correct registration data

### PR DESCRIPTION
**Pull request by cintamani**
From: https://eaflood.atlassian.net/browse/RUBY-515

It makes sure the transient renew registration is populated with the correct data when a renew registration starts with a correct token.

### CODE REVIEW

File: app/models/concerns/waste_exemptions_engine/can_copy_data_from_registration.rb [hunk 1]
```diff
@@ -0,0 +1,54 @@
+# frozen_string_literal: true
+
+module WasteExemptionsEngine
+  module CanCopyDataFromRegistration
```
Comment (irisfaraway) [inline app/models/concerns/waste_exemptions_engine/can_copy_data_from_registration.rb]: Yessss 👏

File: app/models/concerns/waste_exemptions_engine/can_copy_data_from_registration.rb [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    end
+
+    def registration
+      @registration ||= Registration.find_by(reference: reference)
+    end
+
+    private
+
+    def copy_data_from_registration
+      attributes = registration.attributes.except("id",
```
Comment (irisfaraway) [inline app/models/concerns/waste_exemptions_engine/can_copy_data_from_registration.rb]: Does this mean we are copying the old reference to the RenewingRegistration? Is this likely to cause any issues when we complete it?

File: spec/support/helpers/model_properties.rb [hunk 3]
```diff
@@ -51,7 +51,6 @@ module ModelProperties
     TRANSIENT_REGISTRATION_EXEMPTION = (REGISTRATION_EXEMPTION - %i[deregistration_message deregistered_at]).freeze
 
     REGISTRATION = %i[
-      reference
```
Comment (irisfaraway) [inline spec/support/helpers/model_properties.rb]: I'm not sure we should be removing this from the attribute list – it means we are no longer testing Registration as completely in other places like https://github.com/DEFRA/waste-exemptions-engine/blob/0b872a0834ced4a95a75b41ea8fbf172e6bc8e82/spec/models/waste_exemptions_engine/registration_spec.rb#L10-L13

File: app/models/concerns/waste_exemptions_engine/can_copy_data_from_registration.rb (hunk 2 above)
Comment (cintamani) [reply app/models/concerns/waste_exemptions_engine/can_copy_data_from_registration.rb]: No, I have removed the attribute from the one available for copying over :)

Comment (cintamani) [reply app/models/concerns/waste_exemptions_engine/can_copy_data_from_registration.rb]: Is used just for finding it back when we need it :)

**Pull request merged**